android/kotlin/v2/AmpliApp/app/src/main/java/com/example/ampliapp/SegmentDestinationPlugin.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.ampliapp
+
+import android.content.Context
+import com.amplitude.core.platform.DestinationPlugin
+import com.segment.analytics.Analytics
+import com.segment.analytics.Properties
+import com.segment.analytics.Traits
+
+class SegmentDestinationPlugin(appContext: Context, segmentApiKey: String) : DestinationPlugin() {
+    val context: Context = appContext;
+    var analytics: Analytics? = null;
+    val SEGMENT_API_KEY = segmentApiKey;
+
+    override fun setup(amplitude: com.amplitude.core.Amplitude) {
+        super.setup(amplitude)
+        analytics = Analytics.Builder(context, SEGMENT_API_KEY).build()
```

Comment:
Maybe this can be part of the init call? like
```
init {
    analytics = Analytics.Builder(context, SEGMENT_API_KEY).build()
}
```
Then you don't need to define analytics as a var, and no need for others as well.